plugins/eval.py

```diff
@@ -0,0 +1,118 @@
+import base64
+import time
+
+from util import hook, http
+
+
+languages = {
+    "bash": 1,
+    "basic": 3,
+    "c": 4,
+    "cpp": 10,
+    "c++": 10,
+    "csharp": 16,
+    "c#": 16,
+    "clj": 18,
+    "clojure": 18,
+    "crystal": 19,
+    "elixir": 20,
+    "erlang": 21,
+    "go": 22,
+    "haskell": 23,
+    "insect": 25,
+    "java": 26,
+    "js": 29,
+    "javascript": 29,
+    "node": 29,
+    "ocaml": 31,
+    "octave": 32,
+    "pascal": 33,
+    "python": 34,
+    "py": 34,
+    "py3": 34,
+    "python3": 34,
+    "py2": 36,
+    "python2": 36,
+    "ruby": 38,
+    "rb": 38,
+    "rust": 42,
+}
+
+
+def get_result(token):
+    url = "https://api.judge0.com/submissions/{}".format(token)
+
+    try:
+        result = http.get_json(url, get_method="GET")
+    except http.HTTPError as e:
+        return None
+
+    status = result.get("status", {"id": 2})
+    status_id = status["id"]
+
+    # Processing, try again
+    if status_id not in [3, 6, 11]:
```

Comment:
Let's map these status codes to descriptions using (lazily fetched) `https://api.judge0.com/statuses`. There are a lot of other error codes, this should probably be checking `if status_id in [1, 2]` instead (In Queue or Processing).